### PR TITLE
Add support 4.10

### PR DESCRIPTION
**Pull request by alknopfler**
# Description

Preflight check OCP Hub Version (Required Version)
    4.10 - SNO
    4.9 - Compact Cluster
Get Variables automatically with 1 OCP Tag:
Generate these variables from "OC_OCP_VERSION=4.9.7''
        clusterimageset: openshift-v4.9.0
        OC_OCP_TAG: "4.9.0-x86_64"
       OC_RHCOS_RELEASE: "49.84.202110081407-0"

Add Changes to cover 4.10 OCP Version

Fixes # (issue)

## Type of change

Please select the appropriate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [] This change is a documentation update

## Testing

testing in lab cnf12 internally

- [x] Test A -> pipe hub
- [x ] Test B -> pipe spoke

**Test Configuration**:

- Versions: 4.10.5 (ocp) , 2.4 (acm), 4.8(ocs)
- Hardware: cnf12 worker0

## Checklist

- [x] I have performed a self-review of my own code
- [x] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [x] A change should not be merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

**Comment by alknopfler** [comment]
Feat: add support for 4.10 refactoring the spoke.yaml

File: shared-utils/common.sh [hunk 1]
```diff
@@ -162,12 +162,15 @@ if [ ! -f "${SPOKES_FILE}" ]; then
...
+export OC_OCP_VERSION_MIN=${OC_OCP_VERSION_FULL%.*}
+export OC_RHCOS_RELEASE=$(curl -s https://mirror.openshift.com/pub/openshift-v4/clients/ocp/${OC_OCP_VERSION_FULL}/release.txt | grep 'CoreOS' | cut -d' ' -f4)
 export OC_ACM_VERSION=$(yq eval ".config.OC_ACM_VERSION" ${SPOKES_FILE})
 export OC_OCS_VERSION=$(yq eval ".config.OC_OCS_VERSION" ${SPOKES_FILE})
-export OC_OCP_TAG=$(yq eval ".config.OC_OCP_TAG" ${SPOKES_FILE})
-export OC_OCP_VERSION=$(yq eval ".config.OC_OCP_VERSION" ${SPOKES_FILE})
-export CLUSTERIMAGESET=$(yq eval ".config.clusterimageset" ${SPOKES_FILE})
+export OC_OCP_TAG=${OC_OCP_VERSION_FULL}"-x86_64"
+VERSION_WITHOUT_QUOTES="${OC_OCP_VERSION_FULL%\"}"
+VERSION_WITHOUT_QUOTES="${VERSION_WITHOUT_QUOTES#\"}"
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply shared-utils/common.sh:172]: yes, it's just to remove the first quote and the last one  (TBH I could not do in one line)

File: examples/config.yaml [hunk 2]
```diff
@@ -1,8 +1,5 @@
 config:
-  clusterimageset: openshift-v4.9.0
-  OC_OCP_VERSION: "4.9"
-  OC_OCP_TAG: "4.9.0-x86_64"
-  OC_RHCOS_RELEASE: "49.84.202110081407-0" # TODO automate it to get it automated using binary
+  OC_OCP_VERSION: "4.10.6"
   OC_ACM_VERSION: "2.4"
   OC_OCS_VERSION: "4.9"
```
Comment (jparrill) [inline examples/config.yaml:4]: I would put some indicator here to remark that you cannot use (for now) other version of OCS

**Review by jparrill — COMMENTED** [review]
Apart of the comments mentioned, so far so good

**Review by jparrill — DISMISSED** [review]
LGTM

**Review by jparrill — DISMISSED** [review]

**Review by danielchg — APPROVED** [review]
LGTM

**Pull request merged**